projects/igniteui-angular/src/lib/grids/grid-base.component.ts

```diff
@@ -4502,9 +4502,8 @@ export abstract class IgxGridBaseComponent extends DisplayDensityBase implements
      * @hidden
      */
     protected reinitPinStates() {
-        if (this.hasColumnGroups) {
-            this._pinnedColumns = this.columnList.filter((c) => c.pinned);
-        }
+        this._pinnedColumns = this.columnList.filter((c) => c.pinned);
+        
```

Comment:
There are some whitespaces that need to be removed from here. Currently the build fails with the following error: ERROR: projects/igniteui-angular/src/lib/grids/grid-base.component.ts:4510:1 - trailing whitespace